### PR TITLE
Fixed issue #19442: Generated analytics script does not properly escape question group name

### PR DESCRIPTION
**Pull request by mfavetti**
Fixed issue #19442: Generated analytics script does not properly escape question group name

https://bugs.limesurvey.org/view.php?id=19442

master

### CODE REVIEW

File: application/core/LS_Twig_Extension.php [hunk 1]
```diff
@@ -746,7 +746,7 @@ public static function getGoogleAnalyticsTrackingUrl($surveyId, $trackUrlPageNam
             }
         }
 
-        $trackURL = htmlspecialchars($surveyName . '-[' . $surveyId . ']/[' . $page . ']-' . $groupName);
+        $trackURL = htmlspecialchars($surveyName . '-[' . $surveyId . ']/[' . $page . ']-' . $groupName, ENT_QUOTES);
```
[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/core/LS_Twig_Extension.php:749]: > this issue is not related to GA itself at all

OK 

> limesurvey is generating javascript (in this context a string), so the contents of the string must be properly escaped or the browser will not be able to properly parse the javascript. i think it's appropriate to escape the function's entire return value as it's dumped directly into a javascript context later by twig:

Then escape it in twig part ?  No ?

> usage of Chtml::encode vs htmlspecialchars seems to be split 50/50 across code base, is Chtml::encode the preferred method?

I don't know, but i always think it's better to use framework functionnality if we have a framework. Personnal opinion here.

[the remark this answers is not in the document]
Comment (mfavetti) [reply application/core/LS_Twig_Extension.php:749]: ok will escape in twig and update tmrw

[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/core/LS_Twig_Extension.php:749]: > ok will escape in twig and update tmrw

Still my opinion :) Not sure at 100% (i set it as comment, not are "To update")

[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/core/LS_Twig_Extension.php:749]: > ok will escape in twig and update tmrw

Oh ! It's not only an opinion : if someone fix it in template : you broke the fix.

This NEED to be done in template only ! Else : broke API.

**Review by Shnoulle — CHANGES_REQUESTED** [review]
Don't broke API : if someone fix views/subviews/header/google_analytics.twig in extended theme : updating value sent : you broke the theme.

**Review by Shnoulle — APPROVED** [review]

**Pull request merged**